code/modules/vehicles/vehicle.dm

```diff
@@ -0,0 +1,403 @@
+		coeff = A.vehicle_armor[damage_type]
+	return damage*coeff
+
+/obj/vehicle/examine(mob/user)
+	..()
+	var/integrity = health/maxhealth*100
+	switch(integrity)
+		if(85 to 100)
+			user << "It's fully intact."
+		if(65 to 85)
+			user << "It's slightly damaged."
+		if(45 to 65)
+			user << "It's badly damaged."
+		if(25 to 45)
+			user << "It's heavily damaged."
+		else
+			user << "It's falling apart."
+	if(parts)
+		user << "It's has:"
+		for(var/obj/item/part in parts)
+			if(istype(part,/obj/item/weapon/reagent_containers/fueltank))
+				var/obj/item/weapon/reagent_containers/fueltank/check = part
+				var/list/rbeaker = list()
+				for(var/datum/reagent/R in check.reagents.reagent_list)
+					rbeaker += R.name
+				var/contained = english_list(rbeaker)
+				user << "\icon[part] [part] and has [check.reagents.total_volume] out of a max [check.volume], containing [contained]"
+			else
+				user << "\icon[part] [part]"
+
+
+/obj/vehicle/proc/GrantActions(var/mob/living/user)
+	exit_vehicle.vehicle = src
+	exit_vehicle.Grant(user)
+
+/obj/vehicle/proc/RemoveActions(var/mob/living/user)
+	exit_vehicle.Remove(usr)
+
+
+
+/obj/vehicle/proc/exit(var/atom/newloc = loc)
+	RemoveActions()
+	usr.forceMove(newloc)
+	if(usr == driver)
+		driver = null
+	else
+		occupants--
+	usr.reset_view()
+
+/obj/vehicle/Destroy(var/atom/newloc = loc)
+	for(var/mob/M in src)
+		RemoveActions(M)
+		M.forceMove(newloc)
+		M.reset_view()
+	..()
+
+/datum/action/vehicle
+	check_flags = AB_CHECK_RESTRAINED | AB_CHECK_STUNNED | AB_CHECK_ALIVE
+	action_type = AB_INNATE
+	var/obj/vehicle/vehicle
+
+/datum/action/vehicle/exit_vehicle
+	name = "Leave Vehicle"
+	button_icon_state = "mech_eject"
+
+/datum/action/vehicle/exit_vehicle/Activate()
+	if(!owner || !iscarbon(owner))
+		return
+	if(!vehicle)
+		return
+	vehicle.exit()
+
+/obj/vehicle/New()
+	for(var/obj/item/O in parts)
+		O.loc = src
+	var/obj/item/vehicle_parts/engine/E = locate() in parts
+	var/obj/item/weapon/reagent_containers/fueltank/F = locate() in parts
+	F.reagents.add_reagent(E.fueltype, F.volume) //Debug stuff
+	update_stats()
+
+
+
+/obj/vehicle/test/New()
+	parts = newlist(/obj/item/weapon/reagent_containers/fueltank, /obj/item/vehicle_parts/propulsion, /obj/item/vehicle_parts/engine,
+					/obj/item/vehicle_parts/chassis, /obj/item/vehicle_parts/seat, /obj/item/weapon/stock_parts/cell,
+					/obj/item/vehicle_parts/armor)
+	update_stats()
+	..()
+
+
+/obj/vehicle/proc/update_stats()
+	mass = 0 //Reset mass to recalc
+	for(var/obj/item/P in parts)
+		P.loc = src //TRIPLE CHECKS BECAUSE YES
+		var/obj/item/vehicle_parts/O = P
+		if(!O.weight)
+			continue //Ignore massless objects
+		if(O.part_type == "seat")
+			occupants_max += 1
+		O.weight += mass
+	update_icon()
+
+/obj/vehicle/proc/click_action(atom/target,mob/user,params)
+	if(!driver || user.loc != loc )
+		return
+	if(user.incapacitated())
+		return
+	if(!locate(/turf) in list(target,target.loc)) // Prevents inventory from being drilled
```

Comment:
Redundant and not needed. Might be needed in the future if drills are planned, otherwise this can probably be removed.